support/k8s/k8s-workload-registrar/controller.go

```diff
@@ -210,6 +220,44 @@ func (c *Controller) makeID(pathFmt string, pathArgs ...interface{}) string {
 	return id.String()
 }
 
+// Ensure that there is exactly one entry matching these selectors, and it's this one
+func (c *Controller) syncEntry(ctx context.Context, entry *common.RegistrationEntry) error {
+	entries, err := c.c.R.ListBySelectors(ctx, &common.Selectors{
+		Entries: entry.Selectors,
+	})
+	if err != nil {
+		return errs.New("unable to list by pod entries: %v", err)
+	}
+
+	var existing *common.RegistrationEntry = nil
+
+	for _, e := range entries.Entries {
+		if e.ParentId != c.nodeID() {
+			// This is not an entry managed by this registrar
+			continue
+		}
+		if existing == nil {
+			existing = e
+		} else {
+			return errs.New("Multiple pod entries found, with spiffe_id %s and %s", existing.SpiffeId, e.SpiffeId)
+		}
```

Comment:
nit: early return removes some nesting
```suggestion
		if existing != nil {
			return errs.New("Multiple pod entries found, with spiffe_id %s and %s", existing.SpiffeId, e.SpiffeId)
		}
		existing = e
```

Reply:
Done